pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -194,6 +209,9 @@ func (c *ClientCache) FederationClientSetForVersion(version *schema.GroupVersion
 }
 
 func (c *ClientCache) FederationClientForVersion(version *schema.GroupVersion) (*restclient.RESTClient, error) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
 	fedClientSet, err := c.FederationClientSetForVersion(version)
```

Comment:
there is dealock in this PR, for call route:  FederationClientForVersion() -> FederationClientSetForVersion(), the c.mutex.Lock() will be called twice, with no Unlock() being called.